core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -189,13 +190,18 @@ public SparkSqlDialect(SqlDialect.Context context) {
   }
 
   public void unparseSqlIntervalLiteralSpark(SqlWriter writer,
-      SqlIntervalLiteral literal) {
+      SqlIntervalLiteral literal, SqlKind sqlKind) {
     SqlIntervalLiteral.IntervalValue interval =
         (SqlIntervalLiteral.IntervalValue) literal.getValue();
     if (interval.getSign() == -1) {
       writer.print("-");
     }
-    writer.literal(literal.getValue().toString());
+    if (!literal.getValue().toString().equals("1")) {
+      writer.sep("*");
```

Comment:
Please look into this code again, Simple cases like DATE + INTERVAL 'n' DAY will fail after this.